clippy_lints/src/lib.rs

```diff
@@ -1972,6 +1976,7 @@ pub fn register_plugins(store: &mut rustc_lint::LintStore, sess: &Session, conf:
     store.register_late_pass(|| box missing_inline::MissingInline);
     store.register_late_pass(move || box exhaustive_items::ExhaustiveItems);
     store.register_late_pass(|| box if_let_some_result::OkIfLet);
+    store.register_late_pass(|| box while_let_some_result::WhileLetSomeResult);
```

Comment:
cc @rust-lang/clippy do folks think we should just merge these two lints?

Reply:
I think they can at least share a `LintPass`. Merging the lints into one sounds also reasonable. Probably needs a new name then, though.

Reply:
`needless_ok`?

Reply:
Let me know what y'all think about merging! I'll resolve the other PR suggestions for the time being

Reply:
We should absolutely merge the code for the _passes_ either way

Reply:
To clarify, does "merge the code for the `passes`" mean merging `clippy_lints/src/while_let_some_result.rs ` and ` clippy_lints/src/if_let_some_result.rs ` into one? Or does this mean merging the test cases into one? I'm wanting to clarify what is meant by the `passes`

Apologies, still learning the terminologies for this project!

Reply:
Merging the two lint implementations into one. So merging `clippy_lints/src/while_let_some_result.rs ` and `clippy_lints/src/if_let_some_result.rs`. 

The tests should be kept separate if the lints stay separate, but keeping the tests split either way won't hurt.

Reply:
Per discussions, I've gone ahead and merged `clippy_lints/src/while_let_some_result.rs ` and `clippy_lints/src/if_let_some_result.rs` into one file called: `clippy_lints/src/needless_ok.rs` :smile:

Reply:
A couple more ideas. `result_ok` instead of `ok` and `match` instead of `needless`. So `match_result_ok` as in "pattern matching against `Result::ok`".

Reply:
Epic! Let me rename the files in a bit, and I'll make a new comment summarizing these changes at the bottom! 🙌

Reply:
@andrewpollack The idea is to deprecate `if_let_some_result` in favor of a new lint named `match_result_ok`, not just rename the implementation file. But I think we're still waiting for a consensus from the team.

Reply:
That makes sense -- please let me know when consensus is made, and I'm happy to merge them into the new lint!

Reply:
> But I think we're still waiting for a consensus from the team.

@rust-lang/clippy @Manishearth Is everyone okay with the combined `match_result_ok` lint idea?

Reply:
I am!

Reply:
@camsteffen thank you for helping gather consensus! To confirm, instead of two separate lints `while_let_some_result` and `if_let_some_result`, I should combine them into `match_result_ok` with separate actionable descriptions pertaining to the `if` or `while` case?

Reply:
Yes, and mark if_let_some_result as having been renamed to match_result_ok

Reply:
Awesome, I'll ping you both once these changes are made!